File: Python/SSD.py
from Penyimpanan import Penyimpanan

# Derived class dari Penyimpanan (hierarchical inheritance)
class SSD(Penyimpanan):
    def __init__(self, kapasitas_gb=0, interface="", read_speed=0, write_speed=0, 
                 merk="", nama="", harga=0):
        super().__init__(kapasitas_gb, interface, merk, nama, harga)
        self.read_speed = read_speed
        self.write_speed = write_speed
        
    def set_read_speed(self, read_speed):
        self.read_speed = read_speed
        
    def set_write_speed(self, write_speed):
        self.write_speed = write_speed
        
    def get_read_speed(self):
        return self.read_speed
        
    def get_write_speed(self):
        return self.write_speed
        
    def display_info(self):
        super().display_info()
        print(f"Read Speed: {self.read_speed} MB/s")
        print(f"Write Speed: {self.write_speed} MB/s")

File: Java/SSD.java
// Derived class dari Penyimpanan
public class SSD extends Penyimpanan {
    private int readSpeed;  // MB/s
    private int writeSpeed; // MB/s

    public SSD() {
        super();
        this.readSpeed = 0;
        this.writeSpeed = 0;
    }

    public SSD(int kapasitasGB, String interface_, int readSpeed, int writeSpeed, String merk, String nama, int harga) {
        super(kapasitasGB, interface_, merk, nama, harga);
        this.readSpeed = readSpeed;
        this.writeSpeed = writeSpeed;
    }

    public void setReadSpeed(int readSpeed) {
        this.readSpeed = readSpeed;
    }

    public void setWriteSpeed(int writeSpeed) {
        this.writeSpeed = writeSpeed;
    }

    public int getReadSpeed() {
        return this.readSpeed;
    }

    public int getWriteSpeed() {
        return this.writeSpeed;
    }

    @Override
    public void displayInfo() {
        super.displayInfo();
        System.out.println("Read Speed: " + this.readSpeed + " MB/s");
        System.out.println("Write Speed: " + this.writeSpeed + " MB/s");
    }
}
